presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -0,0 +1,227 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.PlanNodeCost;
+import com.facebook.presto.execution.QueryPlan;
+import com.facebook.presto.execution.QueryStats;
+import com.facebook.presto.operator.OperatorStats;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.sql.planner.optimizations.PlanNodeSearcher;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.PlanNodeId;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.DIFFER;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.MATCH;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.NO_BASELINE;
+import static com.facebook.presto.tests.TestTpchDistributedStats.MetricComparison.Result.NO_ESTIMATE;
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunnerWithoutCatalogs;
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+import static org.testng.Assert.assertEquals;
+
+public class TestTpchDistributedStats
+{
+    public static final ImmutableMap<String, Function<PlanNodeCost, Estimate>> METRICS = ImmutableMap.of(
+            "output row count", PlanNodeCost::getOutputRowCount,
+            "output size bytes", PlanNodeCost::getOutputSizeInBytes
+    );
+
+    public static final int NUMBER_OF_TPCH_QUERIES = 22;
+
+    DistributedQueryRunner runner;
+
+    public TestTpchDistributedStats()
+            throws Exception
+    {
+        runner = createQueryRunnerWithoutCatalogs(emptyMap(), emptyMap());
+        runner.createCatalog("tpch", "tpch", ImmutableMap.of(
+                "tpch.column-naming", ColumnNaming.STANDARD.name()
+        ));
+    }
+
+    @Test
+    void testCostEstimatesVsRealityDifferences()
+            throws IOException
+    {
+        IntStream.rangeClosed(1, NUMBER_OF_TPCH_QUERIES)
+                .filter(i -> !asList(15).contains(i)) //query 15 creates a view, which TPCH connector does not support.
+                .forEach(i -> summarizeQuery(i, "EXPLAIN ANALYZE " + getTpchQuery(i)));
```

Comment:
do you need to run `EXPLAIN ANALYZE`? Maybe run plain TPCH query. `EXPLAIN ANALYZE` adds `ExplainAnalyzePlanNode` for which you don't want to compare cost. Also, `OutputNode` will always return 1 row (explain analyze output), which isn't the output of original TPCH query.